### PR TITLE
Add support for reading binary package dependencies.

### PR DESCRIPTION
**Pull request by wlynch**
This change enables support for packages that contain binary dependencies. 

See https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module for the motivation behind this change.

### CODE REVIEW

**Comment by wlynch** [comment]
You're previous comment disappeared from the rebase, but switched over to using GOFLAGS!

We still need some code changes in order to support traversal of binaries. Inspecting the error message feels janky, but I'm not sure how else to ID this type of "error".

Also included an example in the README to show the utility of this.

File: go.mod [hunk 1]
```diff
@@ -7,7 +7,11 @@ require (
 	github.com/google/go-cmp v0.3.0
 	github.com/google/licenseclassifier v0.0.0-20190926221455-842c0d70d702
 	github.com/otiai10/copy v1.0.2
+	github.com/otiai10/curr v0.0.0-20190513014714-f5a3d24e5776 // indirect
 	github.com/spf13/cobra v0.0.5
+	github.com/spf13/pflag v1.0.5 // indirect
+	golang.org/x/net v0.0.0-20191004110552-13f9640d40b9 // indirect
+	golang.org/x/sys v0.0.0-20190801041406-cbf593c0f2f3 // indirect
```
Comment (RJPercival) [inline go.mod]: Did these changes come from running `go get -t ./...`?

File: licenses/library.go [hunk 2]
```diff
@@ -80,9 +80,13 @@ func Libraries(ctx context.Context, classifier Classifier, importPaths ...string
...
-		if len(p.Errors) > 0 {
-			errorOccurred = true
-			return false
+		for _, err := range p.Errors {
+			// Ignore errors complaining that the package is a binary. We still want
+			// to traverse related packages.
+			if !strings.Contains(err.Msg, "is a program, not an importable package") {
+				errorOccurred = true
+				return false
+			}
```
Comment (RJPercival) [inline licenses/library.go]: This seems like it should be a separate PR (all of the other changes are related to build tags, but this seems unrelated).

File: licenses/library.go [hunk 3]
```diff
@@ -80,9 +80,13 @@ func Libraries(ctx context.Context, classifier Classifier, importPaths ...string
 	pkgsByLicense := make(map[string][]*packages.Package)
 	errorOccurred := false
 	packages.Visit(rootPkgs, func(p *packages.Package) bool {
-		if len(p.Errors) > 0 {
-			errorOccurred = true
-			return false
+		for _, err := range p.Errors {
+			// Ignore errors complaining that the package is a binary. We still want
+			// to traverse related packages.
+			if !strings.Contains(err.Msg, "is a program, not an importable package") {
```
Comment (RJPercival) [inline licenses/library.go]: I think this should be controlled by a flag, since I don't think all users will want this behaviour. For my use cases, for example, I only want to include dependencies linked into a binary, not the dependencies of the tooling. What is the your use case that requires this?

File: licenses/library_test.go [hunk 4]
```diff
@@ -69,6 +71,15 @@ func TestLibraries(t *testing.T) {
 				"github.com/google/go-licenses/licenses/testdata/indirect",
 			},
 		},
+		{
+			desc:       "Binary dependency",
```
Comment (RJPercival) [inline licenses/library_test.go]: Thanks for adding this test, it makes it much clearer why you're making this change :)

File: licenses/find.go [hunk 5]
```diff
@@ -42,6 +44,7 @@ func Find(dir string, classifier Classifier) (string, error) {
 	return findUpwards(dir, licenseRegexp, stopAt, func(path string) bool {
 		// TODO(RJPercival): Return license details
 		if _, _, err := classifier.Identify(path); err != nil {
+			glog.Warningf("classifier.Identify(%s): %v", path, err)
```
Comment (RJPercival) [inline licenses/find.go]: This is probably better as Info() logging, since it's rarely indicative of a problem unless you're writing tests and forget to update the classifier stub.

File: licenses/library.go (hunk 3 above)
Comment (wlynch) [reply licenses/library.go]: For our project, we originally we though we were going to need to include things like code generators. After doing some digging, turns out this isn't necessary.

Went ahead an added the flag anyway, since this feels like it might be a useful flag even though I don't have a direct use case for it at the moment. Let me know what you think / if you want to move forward with this. We can always sit and add it later if it comes up.

File: licenses/find.go (hunk 5 above)
Comment (wlynch) [reply licenses/find.go]: Done.

**Comment by AlCutter** [comment]
Seems like the use-case for this went away, I'll close this PR for now.
